quickwit/quickwit-storage/src/split.rs

```diff
@@ -116,13 +117,14 @@ impl SplitPayloadBuilder {
     /// Creates a new SplitPayloadBuilder for given files and hotcache.
     pub fn get_split_payload(
         split_files: &[PathBuf],
+        serialized_split_fields: &[u8],
```

Comment:
You wanted really hard to avoid writing on the file system?

Reply:
Just seemed unnecessary and wasn't much effort. If there's another case, we probably should refactor it to something  like
```rust
enum SplitFile{
   OnDisk(PathBuf),
   InMemory(String, Vec<u8>)
}

Reply:
So if I recall correctly, down the stream, we build a PutPayload off these files.

Would it be possible to carry a PutPayload earlier maybe?